spec/21-http_response_header_format.md

```diff
@@ -0,0 +1,190 @@
+# Trace Context HTTP Response Headers Format
+
+This section describes the binding of the distributed trace context to the `traceresponse` HTTP header.
+
+## Traceresponse Header
+
+The `traceresponse` HTTP response header field identifies a completed request in a tracing system. It has four fields:
+
+* `version`
+* `trace-id`
+* `parent-id`
```

Comment:
I think using "parent" in a response is rather confusing.  Is it referring to the span that initiated the request (and is likely receiving the response), or the span that is sending the response?

Depending on the answer, it should probably be renamed to something else.

Reply:
It is actually neither. The motivational use case for this was the web browser initial page load. The page load happens before the instrumentation js is loaded, so the initial request is sent without a traceparent header. The first instrumented service can then create a "fake" span which it assigns to be it's parent and returns the id of the fake span to the client. The client then uses that id when it creates the span for the page load. So it is the id that is actually assigned to the parent span, making `parent-id` a decent name IMO.

Reply:
I just wanted to chime in that we use response headers at SolarWinds (historically based on X-Trace headers used in Tracelytics, TraceView, and now AppOptics), and our response header includes a trace ID and event ID from the span that wrote the response in the header. So for us "parent" may not strictly be true (it's actually the child), but we can work with any name.

Reply:
Basically, for us, the response header communicates a [happened-before relationship](https://en.wikipedia.org/wiki/Happened-before) between the end of the responding span and the end of the caller span, and also confirms the successful communication of the callee's response, as well as the caller's trace ID, between caller and callee.

Reply:
@cce I think you're describing a substantially different use case.

* "parent" makes sense to communicate back to the browser the ID it _should use for its span_ (because for some reason it could not have created that ID before sending the request to the server)
* @cce's case is to communicate back the server-side event ID

Reply:
parent doesnt make sense. parent passed to a server is the parent of the server span. the span id generated is more often the child of the caller. returning something named parent in the response is confusing and not intuitive. focusing on what you know. ex the span id that serviced the request, is more sensible if this is included at all.

also bear in mind response header processing at all cant be mandatory. this isnt mentioned explicitly. if you do decide to do this it is making this not implementable without significant rewrites of code.

Reply:
@adriancole @cce I think the issue may be that the name of the field does not fully convey it's intended semantic. Given the situation where service `foo` calls service `bar`, if `bar` returns a `parent-id` to foo, it is not saying "this is my span id", but rather "this is the span ID that you, my parent, should use."

The motivating use case is a situation like a browser where the initial request may not contain a span id, but the called service needs to use some id as its "parent," so it generates its own id but also generates an id that it uses as its parent. This parent id is then sent back up the wire to the calling service so that it can either use that id as its own span id, or it can at least link that id to its span.

Would changing the field name to something like `suggested-parent-id`, `assumed-parent-id`, `requested-span-id` or similar be more useful in conveying this semantic?

Reply:
Here is a more visual explanation of the use case:

![image](https://user-images.githubusercontent.com/1612643/74277040-d8191b80-4ce4-11ea-9d70-d99946afeb3c.png)

Reply:
Let's rename this to `proposed-parent-id` (it's not like this string is sent over the wire, a longer name does not hurt).

I would also prefer that the spec clearly states that in case of the caller already providing well-formed `traceparent` header, the `proposed-parent-id` should be omitted.

Reply:
https://github.com/w3c/trace-context/pull/365/files/41aa080c4d1a8c7132a8f8da2170dc861e6461ce..a412b5dd318f10baeab89933eae36e218a1958a8